dll/win32/shell32/brsfolder.cpp

```diff
@@ -6,257 +6,223 @@
-} browse_info;
-
-typedef struct tagTV_ITEMDATA
+    LPBROWSEINFOW    lpBrowseInfo;
+    HWND             hWnd;
+    HWND             hwndTreeView;
+    PIDLIST_ABSOLUTE pidlRet;
+    LAYOUT_DATA*     layout;  // Filled by LayoutInit, used by LayoutUpdate
+    SIZE             szMin;
+    ULONG            hChangeNotify; // Change notification handle
+};
+
+struct BrsItemData
 {
-   LPSHELLFOLDER lpsfParent; /* IShellFolder of the parent */
-   LPITEMIDLIST  lpi;        /* PIDL relative to parent */
-   LPITEMIDLIST  lpifq;      /* Fully qualified PIDL */
-   IEnumIDList*  pEnumIL;    /* Children iterator */ 
-} TV_ITEMDATA, *LPTV_ITEMDATA;
+    CComPtr<IShellFolder>            lpsfParent; // IShellFolder of the parent
+    CComHeapPtr<ITEMIDLIST_RELATIVE> pidlChild;  // PIDL relative to parent
+    CComHeapPtr<ITEMIDLIST_ABSOLUTE> pidlFull;   // Fully qualified PIDL
+    CComPtr<IEnumIDList>             pEnumIL;    // Children iterator
+};
 
 static const LAYOUT_INFO g_layout_info[] =
 {
-    {IDC_BROWSE_FOR_FOLDER_TITLE,         BF_TOP|BF_LEFT|BF_RIGHT},
-    {IDC_BROWSE_FOR_FOLDER_STATUS,        BF_TOP|BF_LEFT|BF_RIGHT},
-    {IDC_BROWSE_FOR_FOLDER_TREEVIEW,      BF_TOP|BF_BOTTOM|BF_LEFT|BF_RIGHT},
-    {IDC_BROWSE_FOR_FOLDER_FOLDER_TEXT,   BF_TOP|BF_LEFT|BF_RIGHT},
-    {IDC_BROWSE_FOR_FOLDER_NEW_FOLDER, BF_BOTTOM|BF_LEFT},
-    {IDOK,              BF_BOTTOM|BF_RIGHT},
-    {IDCANCEL,          BF_BOTTOM|BF_RIGHT}
+    { IDC_BROWSE_FOR_FOLDER_TITLE,       BF_TOP | BF_LEFT | BF_RIGHT },
+    { IDC_BROWSE_FOR_FOLDER_STATUS,      BF_TOP | BF_LEFT | BF_RIGHT },
+    { IDC_BROWSE_FOR_FOLDER_TREEVIEW,    BF_TOP | BF_BOTTOM | BF_LEFT | BF_RIGHT },
+    { IDC_BROWSE_FOR_FOLDER_FOLDER_TEXT, BF_TOP | BF_LEFT | BF_RIGHT },
+    { IDC_BROWSE_FOR_FOLDER_NEW_FOLDER,  BF_BOTTOM | BF_LEFT },
+    { IDOK,                              BF_BOTTOM | BF_RIGHT },
+    { IDCANCEL,                          BF_BOTTOM | BF_RIGHT },
 };
 
-#define LAYOUT_INFO_COUNT (sizeof(g_layout_info)/sizeof(g_layout_info[0]))
-
-#define SUPPORTEDFLAGS (BIF_STATUSTEXT | \
-                        BIF_BROWSEFORCOMPUTER | \
-                        BIF_RETURNFSANCESTORS | \
-                        BIF_RETURNONLYFSDIRS | \
-                        BIF_NONEWFOLDERBUTTON | \
-                        BIF_NEWDIALOGSTYLE | \
-                        BIF_BROWSEINCLUDEFILES)
-
-static void FillTreeView(browse_info*, LPSHELLFOLDER,
-               LPITEMIDLIST, HTREEITEM, IEnumIDList*);
-static HTREEITEM InsertTreeViewItem( browse_info*, IShellFolder *,
-               LPCITEMIDLIST, LPCITEMIDLIST, IEnumIDList*, HTREEITEM);
-
-static inline DWORD BrowseFlagsToSHCONTF(UINT ulFlags)
+#define SUPPORTED_FLAGS (BIF_STATUSTEXT | BIF_BROWSEFORCOMPUTER | BIF_RETURNFSANCESTORS | \
+                         BIF_RETURNONLYFSDIRS | BIF_NONEWFOLDERBUTTON | BIF_NEWDIALOGSTYLE | \
+                         BIF_BROWSEINCLUDEFILES)
+
+static HTREEITEM
+BrsFolder_InsertItem(
```

Comment:
I would suggest `BrsFlder_` instead of `BrsFolder_` . Both: for the prefix and for the filename, because that would allow an 8.3 filename.

Reply:
They are less readable. Readability is over the benefit of `8.3` filename.

Reply:
Renamed as `BrFolder`.

Reply:
Resolved. Thx.